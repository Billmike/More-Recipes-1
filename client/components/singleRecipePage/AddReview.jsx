import React from 'react';
import PropTypes from 'prop-types';

const AddReview = (props) => {
  console.log(props);
  const { reviewContent, handleChange, handleSubmitReview } = props;
  return (
    <div className="row">
      <div className="col s12 m12 l8">
        <form
          className="add-review col s12 center-align"
          onSubmit={handleSubmitReview}
        >
          <div className="row">
            <div type="text" className="input-field col s12">
              <textarea
                id="new-review"
                type="text"
                value={reviewContent}
                className="materialize-textarea"
                onChange={handleChange}
                required
              />
              <label htmlFor="new-review">Add your review</label>
            </div>
          </div>
          <div className="row">
            <button
              type="submit"
              className={`btn waves-effect waves-light indigo accent-2
              white-text`}
            >
              Post Review
            </button>
          </div>
        </form>
      </div>
    </div>
  );
};

AddReview.propTypes = {
  reviewContent: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleSubmitReview: PropTypes.func.isRequired,
};

export default AddReview;
