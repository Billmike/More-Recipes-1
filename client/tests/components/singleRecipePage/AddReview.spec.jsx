import React from 'react';

import AddReview from '../../../components/singleRecipePage/AddReview';

describe('Add review component', () => {
  it('should render component', () => {
    const props = {
      reviewContent: 'This is a nice recipe',
      handleChange: jest.fn(),
      handleAddReview: jest.fn(),
      singleRecipe: {
        isLoading: false,
      }
    };
    const wrapper = shallow(<AddReview {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
  it('should render component with spinner and disabling submit button', () => {
    const props = {
      reviewContent: 'This is a nice recipe',
      handleChange: jest.fn(),
      handleAddReview: jest.fn(),
      singleRecipe: {
        isLoading: true,
      }
    };
    const wrapper = shallow(<AddReview {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper.find('form').length).toEqual(1);
    expect(wrapper.find('Spinner').length).toEqual(1);
    expect(wrapper.find('button').is('[disabled]')).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render component with spinner and disabling submit button', () => {
    const props = {
      reviewContent: 'This is a nice recipe',
      handleChange: jest.fn(),
      handleAddReview: jest.fn(),
      singleRecipe: {
        isLoading: true,
      }
    };

    const wrapper = shallow(<AddReview {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper.find('form').props().onSubmit)
      .toEqual(props.handleAddReview);
    expect(wrapper.find('button').props().type).toEqual('submit');
    expect(wrapper.find('Spinner').length).toEqual(1);
    wrapper.find('textarea').simulate('change', {
      target: { value: 'Wonderful recipe' }
    });
    expect(props.handleChange).toBeCalled();
    expect(wrapper).toMatchSnapshot();
  });
});
