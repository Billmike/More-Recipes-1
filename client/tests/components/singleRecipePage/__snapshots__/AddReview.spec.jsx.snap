// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add review component should render component 1`] = `
<div
  className="row"
>
  <div
    className="col s12 m12 l8"
  >
    <form
      className="add-review col s12 center-align"
      onSubmit={[MockFunction]}
    >
      <div
        className="row"
      >
        <div
          className="input-field col s12"
          type="text"
        >
          <textarea
            className="materialize-textarea"
            id="new-review"
            onChange={[MockFunction]}
            required={true}
            type="text"
            value="This is a nice recipe"
          />
          <label
            htmlFor="new-review"
          >
            Add your review
          </label>
        </div>
      </div>
      <div
        className="row"
      />
      <div
        className="center-align"
      />
      <div
        className="row"
      >
        <button
          className="btn waves-effect waves-light indigo accent-2
              white-text"
          disabled={false}
          id="post-review"
          type="submit"
        >
          Post Review
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`Add review component should render component with spinner and disabling submit button 1`] = `
<div
  className="row"
>
  <div
    className="col s12 m12 l8"
  >
    <form
      className="add-review col s12 center-align"
      onSubmit={[MockFunction]}
    >
      <div
        className="row"
      >
        <div
          className="input-field col s12"
          type="text"
        >
          <textarea
            className="materialize-textarea"
            id="new-review"
            onChange={[MockFunction]}
            required={true}
            type="text"
            value="This is a nice recipe"
          />
          <label
            htmlFor="new-review"
          >
            Add your review
          </label>
        </div>
      </div>
      <div
        className="row"
      />
      <div
        className="center-align"
      >
        <Spinner
          size="small"
        />
      </div>
      <div
        className="row"
      >
        <button
          className="btn waves-effect waves-light indigo accent-2
              white-text"
          disabled={true}
          id="post-review"
          type="submit"
        >
          Post Review
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`Add review component should render component with spinner and disabling submit button 2`] = `
<div
  className="row"
>
  <div
    className="col s12 m12 l8"
  >
    <form
      className="add-review col s12 center-align"
      onSubmit={[MockFunction]}
    >
      <div
        className="row"
      >
        <div
          className="input-field col s12"
          type="text"
        >
          <textarea
            className="materialize-textarea"
            id="new-review"
            onChange={
              [MockFunction] {
                "calls": Array [
                  Array [
                    Object {
                      "target": Object {
                        "value": "Wonderful recipe",
                      },
                    },
                  ],
                ],
              }
            }
            required={true}
            type="text"
            value="This is a nice recipe"
          />
          <label
            htmlFor="new-review"
          >
            Add your review
          </label>
        </div>
      </div>
      <div
        className="row"
      />
      <div
        className="center-align"
      >
        <Spinner
          size="small"
        />
      </div>
      <div
        className="row"
      >
        <button
          className="btn waves-effect waves-light indigo accent-2
              white-text"
          disabled={true}
          id="post-review"
          type="submit"
        >
          Post Review
        </button>
      </div>
    </form>
  </div>
</div>
`;
